html/modules/custom/reliefweb_subscriptions/src/ReliefwebSubscriptionsMailer.php

```diff
@@ -0,0 +1,2142 @@
+    if (!empty($entity->field_source[$language])) {
+      $found = FALSE;
+      foreach ($entity->field_source[$language] as $item) {
+        // No strict equality as tid may be a numeric string.
+        if (isset($item['tid']) && $item['tid'] == 1503) {
+          $found = TRUE;
+          break;
+        }
+      }
+      if (!$found) {
+        return FALSE;
+      }
+    }
+
+    // Check if the format is Situation Report (id: 10).
+    if (!empty($entity->field_content_format[$language])) {
+      $found = FALSE;
+      foreach ($entity->field_content_format[$language] as $item) {
+        // No strict equality as tid may be a numeric string.
+        if (isset($item['tid']) && $item['tid'] == 10) {
+          $found = TRUE;
+          break;
+        }
+      }
+      if (!$found) {
+        return FALSE;
+      }
+    }
+
+    // Check status.
+    if (!empty($entity->is_new)) {
+      $status = $this->getEntityStatus($entity);
+      return in_array($status, ['published', 'to-review']);
+    }
+    elseif (isset($entity->original)) {
+      $status_new = $this->getEntityStatus($entity);
+      $status_old = $this->getEntityStatus($entity->original);
+      return $status_new !== $status_old &&
+        in_array($status_new, ['published', 'to-review']) &&
+        !in_array($status_old, ['published', 'to-review']);
+    }
+    return FALSE;
+  }
+
+  /**
+   * Trigger for disasters.
+   */
+  protected function triggerDisasterNotification($entity, $entity_type) {
+    if (!isset($entity->vocabulary_machine_name) || $entity->vocabulary_machine_name !== 'disaster') {
+      return FALSE;
+    }
+    if (!empty($entity->is_new)) {
+      $status = $this->getEntityStatus($entity);
+      return in_array($status, ['alert', 'current']);
+    }
+    elseif (isset($entity->original)) {
+      $status_new = $this->getEntityStatus($entity);
+      $status_old = $this->getEntityStatus($entity->original);
+      return $status_new !== $status_old && in_array($status_new, [
+        'alert',
+        'current',
+      ]);
+    }
+    return FALSE;
+  }
+
+  /**
+   * Generate sitrep subject.
+   */
+  protected function ochaSitrepSubject($data) {
+    if (!empty($data['title'])) {
+      return 'New OCHA Situation Report - ' . $data['title'];
+    }
+    return '';
+  }
+
+  /**
+   * Generate disaster subject.
+   */
+  protected function disasterSubject($data) {
+    if (!empty($data['name'])) {
+      return 'New Alert/Disaster - ' . $data['name'];
+    }
+    return '';
+  }
+
+  /**
+   * Get the status of an entity.
+   *
+   * @param object $entity
+   *   Entity.
+   *
+   * @return string
+   *   Entity status or empty string.
+   */
+  protected function getEntityStatus($entity) {
+    $language = $entity->language ?? Language::LANGCODE_NOT_SPECIFIED;
+    if (isset($entity->field_status, $entity->field_status[$language][0]['value'])) {
+      return $entity->field_status[$language][0]['value'];
+    }
+    return '';
```

Comment:
The `field_status` is not ported as it's been replaced by D9 workflow states.

Now the status can be retrieved by checking that the `$entity` is an instance of `Drupal\reliefweb_entities\EntityModeratedInterface` and then call `$entity->getModerationStatus()`.